indexer/lib/advertisement-walker.js

```diff
@@ -179,21 +182,26 @@ export async function processNextAdvertisement ({
   } catch (err) {
     let reason
     if (err instanceof Error) {
-      const url = 'url' in err ? err.url : undefined
+      const url = 'url' in err ? err.url : providerInfo.providerAddress
       if ('serverMessage' in err && err.serverMessage) {
         reason = err.serverMessage
+        if ('statusCode' in err && err.statusCode) {
+          reason = `${err.statusCode} ${reason}`
+        }
       } else if ('statusCode' in err && err.statusCode) {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} failed: ${err.statusCode}`
+        reason = err.statusCode
       } else if (err.name === 'TimeoutError') {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} timed out`
+        reason = 'operation timed out'
       } else if (
         err.name === 'TypeError' &&
         err.message === 'fetch failed' &&
         err.cause &&
         err.cause instanceof Error
       ) {
-        reason = `HTTP request to ${url ?? providerInfo.providerAddress} failed: ${err.cause.message}`
+        reason = err.cause.message
       }
+
+      reason = `HTTP request to ${url} failed: ${reason}`
```

Comment:
The changes here are not strictly necessary for this fix. If we were running the proposed version, it would have been much easier for me to discover the actual problem.